packages/react-native-otlp/ios/RNEmbraceOTLP/RNEmbraceOTLP.swift

```diff
@@ -0,0 +1,189 @@
+    var customSpanExporter: OtlpHttpTraceExporter?
+    var customLogExporter: OtlpHttpLogExporter?
+
+    // OTLP HTTP Trace Exporter
+    if let spanConfigDict = spanConfigDict {
+        let config = CustomExporterConfig.parse(spanConfigDict)
+
+        if let spanEndpoint = config.endpoint {
+            let spanTimeout = config.timeout
+            let spanHeaders = config.headers
+
+            customSpanExporter = self.setOtlpHttpTraceExporter(endpoint: spanEndpoint, timeout: spanTimeout, headers: spanHeaders)
+        }
+    }
+
+    // OTLP HTTP Log Exporter
+    if let logConfigDict = logConfigDict {
+        let config = CustomExporterConfig.parse(logConfigDict)
+
+        if let logEndpoint = config.endpoint {
+            let logTimeout = config.timeout
+            let logHeaders = config.headers
+
+            customLogExporter = self.setOtlpHttpLogExporter(endpoint: logEndpoint, timeout: logTimeout, headers: logHeaders)
+        }
+    }
+
+      return OpenTelemetryExport(spanExporter: customSpanExporter, logExporter: customLogExporter)
+    }
+
+    @objc(startNativeEmbraceSDK:otlpExportConfigDict:resolver:rejecter:)
+    func startNativeEmbraceSDK(sdkConfigDict: NSDictionary,
+                               otlpExportConfigDict: NSDictionary,
+                               resolve: @escaping RCTPromiseResolveBlock,
+                               rejecter reject: @escaping RCTPromiseRejectBlock) {
+        let config = SDKConfig(from: sdkConfigDict)
+
+        DispatchQueue.main.async {
+            do {
+                var embraceOptions: Embrace.Options {
+                    var crashReporter: CrashReporter?
+                    if config.disableCrashReporter {
+                        crashReporter = nil
+                    } else {
+                        crashReporter = EmbraceCrashReporter()
+                    }
+
+                    let servicesBuilder = CaptureServiceBuilder().addDefaults()
+                    if config.disableAutomaticViewCapture {
+                        servicesBuilder.remove(ofType: ViewCaptureService.self)
+                    }
+
+                    var endpoints: Embrace.Endpoints?
+                    if config.endpointBaseUrl != nil {
+                        endpoints = Embrace.Endpoints(baseURL: config.endpointBaseUrl!,
+                                                      developmentBaseURL: config.endpointBaseUrl!,
+                                                      configBaseURL: config.endpointBaseUrl!)
+                    }
+
+                    let traceExporter = otlpExportConfigDict.value(forKey: "traceExporter") as? NSDictionary
+                    let logExporter = otlpExportConfigDict.value(forKey: "logExporter") as? NSDictionary
+
+                    if traceExporter == nil && logExporter == nil {
+                        os_log("[Embrace] Neither Traces nor Logs configuration were found, skipping custom export.", log: self.log, type: .info)
```

Comment:
should something different happen in this code path? It appears that `self.setHttpExporters` will still be called in this case, is that fine if both are nil?

Reply:
it is just fine because `OpenTelemetryExport` is handling the case where both are nil. but after all I decided to introduce a tweak since the call of self.setHttpExporters can be avoided if both are not present